device/arista/x86_64-arista_7280cr3mk_32p4/Arista-7280CR3-C40/gearbox_config.json

```diff
@@ -119,224 +119,544 @@
+      "line_tx_fir_main": [13,13,13,13],
+      "line_tx_fir_post1": [-5,-5,-5,-5],
+      "line_tx_fir_post2": [0,0,0,0]
     },
     {
       "name": "Ethernet12",
       "index": 4,
       "phy_id": 1,
       "system_lanes": [2,3],
-      "line_lanes": [16,17,18,19]
+      "line_lanes": [16,17,18,19],
+      "system_tx_fir_pre2": [1,1],
+      "system_tx_fir_pre1": [-5,-5],
+      "system_tx_fir_main": [14,14],
+      "system_tx_fir_post1": [0,0],
+      "system_tx_fir_post2": [0,0],
+      "line_tx_fir_pre2": [0,0,0,0],
+      "line_tx_fir_pre1": [-1,-1,-1,-1],
+      "line_tx_fir_main": [13,13,13,13],
+      "line_tx_fir_post1": [-5,-5,-5,-5],
+      "line_tx_fir_post2": [0,0,0,0]
     },
     {
       "name": "Ethernet16",
       "index": 5,
       "phy_id": 2,
       "system_lanes": [4,5],
-      "line_lanes": [20,21,22,23]
+      "line_lanes": [20,21,22,23],
+      "system_tx_fir_pre2": [1,1],
+      "system_tx_fir_pre1": [-5,-5],
+      "system_tx_fir_main": [14,14],
+      "system_tx_fir_post1": [0,0],
+      "system_tx_fir_post2": [0,0],
+      "line_tx_fir_pre2": [0,0,0,0],
+      "line_tx_fir_pre1": [-1,-1,-1,-1],
+      "line_tx_fir_main": [13,13,13,13],
+      "line_tx_fir_post1": [-5,-5,-5,-5],
+      "line_tx_fir_post2": [0,0,0,0]
     },
     {
       "name": "Ethernet20",
       "index": 6,
       "phy_id": 2,
       "system_lanes": [6,7],
-      "line_lanes": [8,9,10,11]
+      "line_lanes": [8,9,10,11],
+      "system_tx_fir_pre2": [1,1],
+      "system_tx_fir_pre1": [-5,-5],
+      "system_tx_fir_main": [14,14],
+      "system_tx_fir_post1": [0,0],
+      "system_tx_fir_post2": [0,0],
+      "line_tx_fir_pre2": [0,0,0,0],
+      "line_tx_fir_pre1": [-1,-1,-1,-1],
+      "line_tx_fir_main": [13,13,13,13],
+      "line_tx_fir_post1": [-5,-5,-5,-5],
+      "line_tx_fir_post2": [0,0,0,0]
     },
     {
       "name": "Ethernet24",
       "index": 7,
       "phy_id": 2,
       "system_lanes": [0,1],
-      "line_lanes": [12,13,14,15]
+      "line_lanes": [12,13,14,15],
+      "system_tx_fir_pre2": [1,1],
+      "system_tx_fir_pre1": [-5,-5],
+      "system_tx_fir_main": [14,14],
+      "system_tx_fir_post1": [0,0],
+      "system_tx_fir_post2": [0,0],
+      "line_tx_fir_pre2": [0,0,0,0],
+      "line_tx_fir_pre1": [-1,-1,-1,-1],
+      "line_tx_fir_main": [13,13,13,13],
+      "line_tx_fir_post1": [-5,-5,-5,-5],
+      "line_tx_fir_post2": [0,0,0,0]
     },
     {
       "name": "Ethernet28",
       "index": 8,
       "phy_id": 2,
       "system_lanes": [2,3],
-      "line_lanes": [16,17,18,19]
+      "line_lanes": [16,17,18,19],
+      "system_tx_fir_pre2": [1,1],
+      "system_tx_fir_pre1": [-5,-5],
+      "system_tx_fir_main": [14,14],
+      "system_tx_fir_post1": [0,0],
+      "system_tx_fir_post2": [0,0],
+      "line_tx_fir_pre2": [0,0,0,0],
+      "line_tx_fir_pre1": [-1,-1,-1,-1],
+      "line_tx_fir_main": [13,13,13,13],
+      "line_tx_fir_post1": [-5,-5,-5,-5],
+      "line_tx_fir_post2": [0,0,0,0]
```

Comment:
The line side SI settings may depend upon the type of optics for some platforms. In that case, Xcvrd should read these values based upon the media ([see Ref](https://github.com/Azure/sonic-platform-daemons/blob/master/sonic-xcvrd/xcvrd/xcvrd.py#L639)) and publish to gearbox syncd to program the line side on the gearbox. Can we discuss this scheme in the community?

Reply:
@prgeor , this gearbox_config.json here is just to provide some initial value like setting tuning values for brcm ASIC in brcm config file. For the dynamic tuning support (based on xcvr type), an idea is to add the support in the future with some extension to the usage of media_settings.json.